src/metadata/Metadata.ts

```diff
@@ -297,6 +299,69 @@ export class Metadata extends Instantiable {
         return result
     }
 
+    /**
+     * Retrieves a DDO by DID.
+     * @param  {string} agreementId agreementId of the asset.
+     * @return {Promise<Service>} DDO of the asset.
+     */
+    public async retrieveServiceAgreement(
+        agreementId: string,
+        metadataServiceEndpoint?: string
+    ): Promise<Service> {
+        const fullUrl =
+            metadataServiceEndpoint || `${this.url}${agreementPath}/${agreementId}`
+        const result = await this.nevermined.utils.fetch
+            .get(fullUrl)
+            .then((response: any) => {
+                if (response.ok) {
+                    return response.json()
+                }
+                this.logger.log(
+                    'retrieveDDO failed:',
+                    response.status,
+                    response.statusText,
+                    agreementId
+                )
+                return null as Service
+            })
+            .then((response: Service) => {
+                return response as Service
+            })
+            .catch(error => {
+                this.logger.error('Error retrieving metadata: ', error)
+                return null as Service
+            })
+
+        return result
+    }
+
+    public async storeServiceAgreement(agreement: Service): Promise<Service> {
+        const fullUrl = `${this.url}${agreementPath}`
+        const result: Service = await this.nevermined.utils.fetch
+            .post(fullUrl, JSON.stringify(agreement))
+            .then((response: any) => {
+                if (response.ok) {
+                    return response.json()
+                }
+                this.logger.error(
+                    'storeServiceAngrement failed:',
```

Comment:
typo

Reply:
I just had lunch. I'll start on this now 👍